### PR TITLE
Fix invalid default shipdesigns

### LINKED ISSUE #1663: Invalid ShipDesign Warnings
Observed in the master build server log:
````12:49:35.684842 [debug] server : ShipDesign.cpp:1290 : Initializing PredefinedShipDesignManager
12:49:35.717862 [warn] server : ShipDesign.cpp:1026 : Invalid ShipDesign part FU_N_DIMENSIONAL_ENGINE_MATRIX conflicts with "FU_SINGULARITY_ENGINE_CORE". Removing "FU_N_DIMENSIONAL_ENGINE_MATRIX"
12:49:35.718863 [warn] server : ShipDesign.cpp:1083 : Invalid ShipDesign:
ShipDesign
    name = "SD_GRAVITATING2"
    uuid = "08a58b08-0929-496d-84fc-faa91424ca20"
    description = "SD_GRAVITATING2_DESC"
    hull = "SH_SELF_GRAVITATING"
    parts = [
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_0_1"
        "FU_SINGULARITY_ENGINE_CORE"
        "AR_XENTRONIUM_PLATE"
        "FU_N_DIMENSIONAL_ENGINE_MATRIX"
        "SH_BLACK"
    ]
    model = "mark1"

ShipDesign was made valid as:
ShipDesign
    name = "SD_GRAVITATING2"
    uuid = "08a58b08-0929-496d-84fc-faa91424ca20"
    description = "SD_GRAVITATING2_DESC"
    hull = "SH_SELF_GRAVITATING"
    parts = [
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_0_1"
        "FU_SINGULARITY_ENGINE_CORE"
        "AR_XENTRONIUM_PLATE"
        ""
        "SH_BLACK"
    ]
    model = "mark1"


12:49:35.747896 [warn] server : ShipDesign.cpp:1026 : Invalid ShipDesign part FU_N_DIMENSIONAL_ENGINE_MATRIX conflicts with "FU_N_DIMENSIONAL_ENGINE_MATRIX". Removing "FU_N_DIMENSIONAL_ENGINE_MATRIX"
12:49:35.747896 [warn] server : ShipDesign.cpp:1026 : Invalid ShipDesign part FU_SINGULARITY_ENGINE_CORE conflicts with "FU_N_DIMENSIONAL_ENGINE_MATRIX". Removing "FU_SINGULARITY_ENGINE_CORE"
12:49:35.747896 [warn] server : ShipDesign.cpp:1083 : Invalid ShipDesign:
ShipDesign
    name = "SD_ROBO_TITAN1"
    uuid = "08a58b08-0929-496d-84fc-faa91424ca19"
    description = "SD_ROBO_TITAN1_DESC"
    hull = "SH_TITANIC"
    parts = [
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "AR_XENTRONIUM_PLATE"
        "AR_XENTRONIUM_PLATE"
        "AR_XENTRONIUM_PLATE"
        "AR_XENTRONIUM_PLATE"
        "SH_PLASMA"
        "FU_N_DIMENSIONAL_ENGINE_MATRIX"
        "FU_N_DIMENSIONAL_ENGINE_MATRIX"
        "FU_SINGULARITY_ENGINE_CORE"
    ]
    model = "mark1"

ShipDesign was made valid as:
ShipDesign
    name = "SD_ROBO_TITAN1"
    uuid = "08a58b08-0929-496d-84fc-faa91424ca19"
    description = "SD_ROBO_TITAN1_DESC"
    hull = "SH_TITANIC"
    parts = [
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "SR_WEAPON_4_1"
        "AR_XENTRONIUM_PLATE"
        "AR_XENTRONIUM_PLATE"
        "AR_XENTRONIUM_PLATE"
        "AR_XENTRONIUM_PLATE"
        "SH_PLASMA"
        "FU_N_DIMENSIONAL_ENGINE_MATRIX"
        ""
        ""
    ]
    model = "mark1"
````

### PR DESCRIPTION
**Pull request by dbenage-cx**
Two default shipdesigns were not caught with the change of excluding engines.

Replaces the second engine in gravitating hull with a fuel tank.
Replaces the two excess engines(and two direct weapons) in titanic hull with fighters.

Fixes #1663

### CODE REVIEW

File: default/scripting/ship_designs/SD_GRAVITATING2.focs.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ ShipDesign
         "SR_WEAPON_0_1"
         "FU_SINGULARITY_ENGINE_CORE"
         "AR_XENTRONIUM_PLATE"
-        "FU_N_DIMENSIONAL_ENGINE_MATRIX"
+        "FU_DEUTERIUM_TANK"
```
Comment (MatGB) [inline default/scripting/ship_designs/SD_GRAVITATING2.focs.txt]: Not sure about Dueterium, currently fuel is barely used and last we asked no one researches the techs, supply is on my long term list to balance but perhaps just extra fuel tank?

File: default/scripting/ship_designs/SD_ROBO_TITAN1.focs.txt [hunk 2]
```diff
@@ -12,17 +12,17 @@ ShipDesign
...
-        "SR_WEAPON_4_1"
-        "SR_WEAPON_4_1"
-        "SR_WEAPON_4_1"
-        "SR_WEAPON_4_1"
         "AR_XENTRONIUM_PLATE"
         "AR_XENTRONIUM_PLATE"
+        "FT_BAY_1"
+        "SR_WEAPON_4_1"
+        "SR_WEAPON_4_1"
+        "FT_BAY_1"
```
Comment (MatGB) [inline default/scripting/ship_designs/SD_ROBO_TITAN1.focs.txt:20]: Yup, exactly my design, was sure I'd done this in master as well.

**Review by MatGB — COMMENTED** [review]
Close to what I'd have done (and thought I had), the deuterium tank would make sense but I don't think anyone ever researches fuel so it won't show up as a sample design, the basic tank would probably make more sense but up to you.

**Comment by dbenage-cx** [comment]
Ok, changed to basic tank.  TY for review, will merge later if no other concerns.

**Pull request merged**